contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +172,47 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.  This token is used for both discovery and TLS bootstrap.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
+
+### `kubeadm join` flags
+
+* `--token` This sets the token for both discovery and bootstrap auth.
+* `--discovery-url` If set this will grab the cluster-info data (a kubeconfig) from a URL. Due to the sensitive nature of this data, we will only support https URLs.  This also supports `username:password@host` syntax for doing HTTP auth.
+* `--discovery-file` If set, this will load the cluster-info from a file.
+* `--discovery-token` If set, (or set via `--token`) then we will be using the token scheme described above.
```

Comment:
I'm confused here.

Discovery == "getting and trusting the cluster-info file".  We have 3 ways to do this: token, HTTPS endpoint (trusting the system CA bundle) and local file.  This is driven by the `--discovery-*` flags.

I think you are wrong wrt 2.  The TLS bootstrap dance only works with a token.  It is what is used to authenticate to the API server using temporary credentials (the token) in order to submit a CSR. That would be the `--tls-bootstrap-token` flag. I'll make sure we have that.